src/v0/destinations/intercom/deleteUsers.js

```diff
@@ -11,35 +11,38 @@ const userDeletionHandler = async (userAttributes, config) => {
     throw new CustomError("api key for deletion not present", 400);
   }
 
-  for (let i = 0; i < userAttributes.length; i += 1) {
-    const uId = userAttributes[i].userId;
-    if (!uId) {
-      throw new CustomError("User id for deletion not present", 400);
-    }
-    const requestOptions = {
-      method: "delete",
-      url: `https://api.intercom.io/contacts/${uId}`,
-      headers: {
-        Authorization: `Bearer ${apiKey}`
+  await Promise.all(
+    userAttributes.map(async ua => {
+      const uId = ua.userId;
+      if (!uId) {
+        throw new CustomError("User id for deletion not present", 400);
       }
-    };
-    const resp = await httpSend(requestOptions);
-    if (!resp || !resp.response) {
-      throw new CustomError("Could not get response", 500);
-    }
-    if (
-      resp &&
-      resp.response &&
-      resp.response.response &&
-      resp.response.response.status !== 200 &&
-      resp.response.response.status !== 404 // this will be returned if user is not found. Will send successfull to server
-    ) {
-      throw new CustomError(
-        resp.response.response.statusText || "Error while deleting user",
-        resp.response.response.status
-      );
-    }
-  }
+      const requestOptions = {
+        method: "delete",
+        url: `https://api.intercom.io/contacts/${uId}`,
+        headers: {
+          Authorization: `Bearer ${apiKey}`
+        }
+      };
+      const resp = await httpSend(requestOptions);
+      if (!resp || !resp.response) {
+        throw new CustomError("Could not get response", 500);
+      }
+      if (
+        resp &&
+        resp.response &&
+        resp.response.response &&
+        resp.response.response.status !== 200 &&
+        resp.response.response.status !== 404 // this will be returned if user is not found. Will send successful to server
```

Comment:
resp?.response?.response?.status &&  resp.response.response.status !== 200 && resp.response.response.status !== 404

Reply:
This has been addressed. Hence, resolving it.